library/Fission/Internal/UTF8.hs

```diff
@@ -27,20 +42,51 @@ instance Textable ByteString where
 instance Textable Lazy.ByteString where
   encode = encode . Lazy.toStrict
 
+
+
+-- Functions
+
+
 showLazyBS :: Show a => a -> Lazy.ByteString
 showLazyBS = textToLazyBS . textDisplay . displayShow
 
 textToLazyBS :: Text -> Lazy.ByteString
 textToLazyBS = Lazy.fromStrict . Text.encodeUtf8
 
+{-| Strip one newline character from the end of a lazy `ByteString`.
+
+    >>> stripNewline ";)\n"
+    ";)"
+
+    >>> stripNewline "<>\n\n"
+    "<>\n"
+
+    prop> stripNewline (Lazy.append bs "\n") == bs
```

Comment:
Maybe another prop test that it doesn't do multiple newlines, too:

```haskell
prop> stripNewline (Lazy.append bs "\n\n") /= bs
```

Reply:
🤔 Where doctests really shine is in story telling:

```haskell
{-| Strip one newline character from the end of a lazy `ByteString`
      
    It removes a newline from the end of strings:  
  
    >>> stripNewline ";)\n"
    ";)"

    prop> stripNewline (Lazy.append bs "\n") == bs

    But only does a single newline:

    >>> stripNewline "<>\n\n"
    "<>\n"

    prop> stripNewline (Lazy.append bs "\n\n") == (bs <> "\n")

    It also doesn't touch other types of whitespace:

    prop> stripNewline (Lazy.append bs "\r") == (bs <> "\r")

    Or newlines at the beginning of the string:

    prop> stripNewline (Lazy.append "\n" bs) == ("\n" <> bs)
-}
```

Reply:
Actually, you know what I take that back. These don't really add to the _docs_, but are good test cases for the actual test suite. Leaving here instead of deleting because... sure why not.

Reply:
Will add the property test 👍 
This type of story telling could be great for more complicated functions though?

Reply:
> could be great for more complicated functions though?

Yeah, I think that it's great in general, and is how I tend to write doctests. I also realize that it can make the docs pretty noisy, so it's a balance.